nomad/structs/services.go

```diff
@@ -896,11 +915,48 @@ func (s *ConsulSidecarService) Copy() *ConsulSidecarService {
 	if s == nil {
 		return nil
 	}
+
+	var checks []*ServiceCheck
+	if s.Checks != nil {
+		checks = make([]*ServiceCheck, len(s.Checks))
+		for i, c := range s.Checks {
+			checks[i] = c.Copy()
+		}
+	}
+
 	return &ConsulSidecarService{
-		Tags:  helper.CopySliceString(s.Tags),
-		Port:  s.Port,
-		Proxy: s.Proxy.Copy(),
+		Tags:   helper.CopySliceString(s.Tags),
+		Port:   s.Port,
+		Proxy:  s.Proxy.Copy(),
+		Checks: checks,
+	}
+
+}
+
+func (s *ConsulSidecarService) Canonicalize() {
+	if len(s.Checks) == 0 {
+		s.Checks = nil
+	}
+
+	for _, check := range s.Checks {
+		check.Canonicalize("sidecar_service")
+	}
+}
+
+func (s *ConsulSidecarService) Validate() error {
+	var mErr multierror.Error
+	for _, c := range s.Checks {
+
+		//if c.PortLabel != "" {
+		//	mErr.Errors = append(mErr.Errors, fmt.Errorf("Check %s invalid: port is automatically assigned", c.Name))
+		//	continue
+		//}
```

Comment:
Not sure if we should allow setting port labels here; as the sidecar service doesn't specify ports.